snowflake/src/main/java/org/apache/iceberg/snowflake/SnowflakeCatalog.java

```diff
@@ -109,6 +110,10 @@ public void initialize(String name, Map<String, String> properties) {
               + " JDBC driver to your jars/packages",
           cnfe);
     }
+
+    // Populate application identifier in jdbc client
+    properties.put("application", APP_IDENTIFIER);
```

Comment:
we're using `CatalogProperties.APP_ID` in a few other places, would it make sense to use it here as well or does the JDBC client require `application`?

Reply:
I feel like `APP_ID` would be the identifier of where the catalog is being used (e.g. the spark application id) not something specific to this particular catalog implementation.

Reply:
APP_ID property is not supported by Snowflake's jdbc driver, it requires application. Also as Daniel pointed out, application property does not uniquely point to each catalog usage but broadly identifies the partner app type.